### PR TITLE
feat: add sample code for multiple inequalities indexing consideration query

### PR DESCRIPTION
**Pull request by cindy-peng**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-datastore/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

File: samples/snippets/src/main/java/com/example/datastore/filters/OrderFieldsQuery.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    Query<Entity> query =
+            Query.newEntityQueryBuilder()
+                    .setKind("employees")
+                    .setFilter(PropertyFilter.gt("salary", 100000))
+                    .setOrderBy(OrderBy.asc("salary"))
+                    .build();
+
+    // Get the results back from Datastore
+    QueryResults<Entity> results = datastore.run(query);
+    // Order results by `experience`
```
Comment (daniel-sanche) [inline samples/snippets/src/main/java/com/example/datastore/filters/OrderFieldsQuery.java:50]: should there be some code after this comment?

File: samples/snippets/src/main/java/com/example/datastore/filters/IndexingConsiderationQuery.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import com.google.cloud.datastore.StructuredQuery.PropertyFilter;
+
+public class IndexingConsiderationQuery {
+  public static void invoke() throws Exception {
+
+    // Instantiates a client
+    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+
+    // Build a query with multi inequal filters and optimized index order of index properties.
+    // [START datastore_query_indexing_considerations]
```
Comment (daniel-sanche) [inline samples/snippets/src/main/java/com/example/datastore/filters/IndexingConsiderationQuery.java:41]: Note: the best practice is for samples to be copy/pasteable, meaning the client and imports are included in the region tag

But IIRC datastore doesn't really follow that convention, so if that's the case, we can just consider that in the future

**Review by daniel-sanche — APPROVED** [review]
LGTM, but take a look at my comment on `// Order results by experience`

File: samples/snippets/src/main/java/com/example/datastore/filters/OrderFieldsQuery.java (hunk 1 above)
Comment (cindy-peng) [reply samples/snippets/src/main/java/com/example/datastore/filters/OrderFieldsQuery.java:50]: Oh I was assuming users can have the choices of their own to use different approaches for sorting. This sorting logic wasn't provided in the sample on the doc site either: https://cloud.google.com/datastore/docs/multiple-range-fields#java_2
Should we be consistent here?

File: samples/snippets/src/main/java/com/example/datastore/filters/IndexingConsiderationQuery.java (hunk 2 above)
Comment (cindy-peng) [reply samples/snippets/src/main/java/com/example/datastore/filters/IndexingConsiderationQuery.java]: Good point. I do see in some of the datastore samples, the client and imports were included in the region tag. I have updated the new region tags to include those.

**Pull request merged**